hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -0,0 +1,153 @@
+        fields.add(new HCatFieldSchema(name, getType(name, field.getSchema().getType()), ""));
+      } catch (HCatException e) {
+        LOG.error("Failed to create HCatFieldSchema field {} of type {} from schema", name,
+                  field.getSchema().getType());
+      }
+    }
+    return new HCatSchema(fields);
+  }
+
+  /**
+   * Returns the {@link PrimitiveTypeInfo} for the {@link Schema.Type}
+   *
+   * @param name name of the field
+   * @param category {@link Schema.Type} of the field
+   * @return {@link PrimitiveTypeInfo} for the given {@link Schema.Type} which is compatible with Hive.
+   */
+  private static PrimitiveTypeInfo getType(String name, Schema.Type category) {
+    switch (category) {
+      case BOOLEAN:
+        return TypeInfoFactory.booleanTypeInfo;
+      case INT:
+        return TypeInfoFactory.intTypeInfo;
+      case LONG:
+        return TypeInfoFactory.longTypeInfo;
+      case FLOAT:
+        return TypeInfoFactory.floatTypeInfo;
+      case DOUBLE:
+        return TypeInfoFactory.doubleTypeInfo;
+      case STRING:
+        return TypeInfoFactory.stringTypeInfo;
+      case BYTES:
+        return TypeInfoFactory.binaryTypeInfo;
+      default:
+        throw new IllegalArgumentException(String.format(
+          "Schema contains field '%s' with unsupported type %s", name, category.name()));
+    }
+  }
+
+  /**
+   * <p>Converts a {@link HCatSchema} from hive to {@link Schema} for CDAP.</p>
+   * <p><b>Note:</b> This conversion does not support non-primitive types and the conversion will fail.
+   * The conversion might also change the primitive type.
+   * See {@link #getType(String, PrimitiveObjectInspector.PrimitiveCategory)} for details.</p>
+   *
+   * @param hiveSchema the {@link HCatSchema} of the hive table
+   * @return {@link Schema} for the given {@link HCatSchema}
+   */
+  public static Schema toSchema(HCatSchema hiveSchema) {
+    List<Schema.Field> fields = Lists.newArrayList();
+    for (HCatFieldSchema field : hiveSchema.getFields()) {
+      String name = field.getName();
+      if (field.isComplex()) {
+        throw new IllegalArgumentException(String.format(
+          "Table schema contains field '%s' with complex type %s. Only primitive types are supported.",
+          name, field.getTypeString()));
+      }
+      fields.add(Schema.Field.of(name, Schema.of(getType(name, field.getTypeInfo().getPrimitiveCategory()))));
+    }
+    return Schema.recordOf("record", fields);
+  }
+
+  /**
+   * Returns the {@link Schema.Type} compatible for this field from hive.
+   *
+   * @param name name of the field
+   * @param category the field's {@link PrimitiveObjectInspector.PrimitiveCategory}
+   * @return the {@link Schema.Type} for this field
+   */
+  private static Schema.Type getType(String name, PrimitiveObjectInspector.PrimitiveCategory category) {
+    switch (category) {
+      case BOOLEAN:
+        return Schema.Type.BOOLEAN;
```

Comment:
don't remember if hive types have a concept of nullable or not.  If so, the schema returned should reflect that. If not, it seems like we should always return a nullable type.